### PR TITLE
Thread keep alive feature

### PR DESCRIPTION
**Pull request by Jlobblet**
Adds a feature to the Watcher that allows it to keep a thread alive indefinitely. This is done by checking whether the thread is already archived and unarchiving it, or sending a message in the channel to keep it alive.

This change adds two new config fields, specifying the IDs of which threads to preserve and a message to send.

https://github.com/Jlobblet/WatcherBot/blob/532002ac59a1d88ae98992eabaf8de88630ee5b4/WatcherBot/appsettings.example.json#L102-L105

In addition, this change bumps the versions of most packages and cleans up some depcreciated code and warnings. The bumps were necessary because features in later DisCatSharp releases were required, specifically the ability to always fetch a thread from the API rather than relying on the cache. This is required to track the most recent message in the thread reliably.

### CODE REVIEW

File: WatcherBot/Config/Config.cs [hunk 1]
```diff
@@ -9,7 +9,7 @@ namespace WatcherBot.Config;
 public class Config
 {
     public const string ConfigSection = "Watcher";
-    private readonly Lazy<Dictionary<ulong, Range>> attachmentLimitsLazy;
+    private readonly System.Lazy<Dictionary<ulong, Range>> attachmentLimitsLazy;
```
Comment (Saalvage) [inline WatcherBot/Config/Config.cs:12]: Why `System.`?

Comment (Jlobblet) [reply WatcherBot/Config/Config.cs:12]: I don't know where this change originated from. I blame my IDE.

**Review by juanjp600 — APPROVED** [review]
Everything here looks fine to me :+1:

File: WatcherBot/appsettings.example.json [hunk 2]
```diff
@@ -99,6 +99,10 @@
         "Networking": "Cyan"
       }
     },
+    "KeepAliveThreadIds": [
+      0
+    ],
+    "KeepAliveMessage": "Ah, ha, ha, ha, stayin' alive, stayin' alive",
```
Comment (Saalvage) [inline WatcherBot/appsettings.example.json:105]: Watcher bot would never say this I'm crying right now

File: WatcherBot/Utils/BarotraumaToolBox.cs [hunk 3]
```diff
@@ -135,4 +164,13 @@ public static async Task ReportSpam(BotMain botMain, DiscordMessage spamMessage,
...
     }
+
+    public static TimeSpan ToTimeSpan(this ThreadAutoArchiveDuration duration) => duration switch
+    {
+        ThreadAutoArchiveDuration.OneHour   => TimeSpan.FromHours(1),
+        ThreadAutoArchiveDuration.OneDay    => TimeSpan.FromDays(1),
+        ThreadAutoArchiveDuration.ThreeDays => TimeSpan.FromDays(3),
+        ThreadAutoArchiveDuration.OneWeek   => TimeSpan.FromDays(7),
+        _                                   => throw new ArgumentOutOfRangeException(nameof(duration), duration, null),
+    };
```
[the remark this answers is not in the document]
Comment (Jlobblet) [reply WatcherBot/Utils/BarotraumaToolBox.cs]: Addressed in [d0ca095](https://github.com/juanjp600/WatcherBot/pull/27/commits/d0ca0959ad772f743e095fa3cb4d6eba39105337)

File: WatcherBot/appsettings.example.json (hunk 2 above)
Comment (Jlobblet) [reply WatcherBot/appsettings.example.json:105]: It's just the example

**Review by Saalvage — APPROVED** [review]
Everything here looks fine to me 👍

**Pull request merged**